java/pom.xml

```diff
@@ -53,20 +53,20 @@
       <subscribe>dev-subscribe@arrow.apache.org</subscribe>
       <unsubscribe>dev-unsubscribe@arrow.apache.org</unsubscribe>
       <post>dev@arrow.apache.org</post>
-      <archive>http://mail-archives.apache.org/mod_mbox/arrow-dev/</archive>
+      <archive>https://mail-archives.apache.org/mod_mbox/arrow-dev/</archive>
```

Comment:
why is that needed ?

Reply:
You're right, this is not needed, event if the good url is with the https. Shall I revert ?

Reply:
I think it's better to leave it as it is originally, this way we have less changes.